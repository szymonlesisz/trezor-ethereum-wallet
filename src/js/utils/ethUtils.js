/* @flow */


import BigNumber from 'bignumber.js';

export const decimalToHex = (dec: number): string => new BigNumber(dec).toString(16);

export const hexToDecimal = (hex: number): string => {
    const sanitized: ?string = sanitizeHex(hex);
    return !sanitized ? 'null' : new BigNumber(sanitized).toString();
};

export const sanitizeHex = (hex: number | string): ?string => {
    if (typeof hex !== 'string') return null;
    hex = hex.substring(0, 2) === '0x' ? hex.substring(2) : hex;
    if (hex === '') return '';
    return `0x${padLeftEven(hex)}`;
};

export const padLeftEven = (hex: string): string => {
    hex = hex.length % 2 != 0 ? `0${hex}` : hex;
    return hex;
};

export const strip = (str: string): string => {
    if (str.indexOf('0x') === 0) {
        return padLeftEven(str.substring(2, str.length));
    }
    return padLeftEven(str);
};

export const calcGasPrice = (price: BigNumber, limit: string): string => price.times(limit).toString();