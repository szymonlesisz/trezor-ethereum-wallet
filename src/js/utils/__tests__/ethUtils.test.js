import * as ethUtils from '../ethUtils';

describe('eth utils', () => {
    it('decimalToHex', () => {
        const input = [0, 1, 2, 100, 999];

        input.forEach((entry) => {
            expect(ethUtils.decimalToHex(entry)).toMatchSnapshot();
        });
    });
});
