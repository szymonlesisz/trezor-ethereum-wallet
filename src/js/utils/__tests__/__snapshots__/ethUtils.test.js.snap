// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eth utils decimalToHex 1`] = `"0"`;

exports[`eth utils decimalToHex 2`] = `"1"`;

exports[`eth utils decimalToHex 3`] = `"2"`;

exports[`eth utils decimalToHex 4`] = `"64"`;

exports[`eth utils decimalToHex 5`] = `"3e7"`;
